core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -11270,40 +11323,48 @@ private AuthenticationResult authenticateInternalWithID(String preferredUserName
                         .getAttributeName(getMyDomainName(), preferredUserNameClaim);
                 // Let's authenticate with the primary UserStoreManager.
 
-                if (abstractUserStoreManager.isUniqueUserIdEnabled()) {
-                    authenticationResult = abstractUserStoreManager
-                            .doAuthenticateWithID(preferredUserNameProperty, preferredUserNameValue, credentialObj,
-                                    profileName);
+                if (abstractUserStoreManager.isCircuitBreakerOpen()) {
```

Comment:
why this `userStoreManager.isCircuitBreakerOpen()` state is checked only in user listing and authentication?

Why it's not checked on single-user retrieval flow?

Reply:
Could you elaborate on this a bit more?

Reply:
@AnuradhaSK to add to the previous comment, as we are concerned about addressing the authentication failure flow due to $subject, the above-mentioned flow is not a part of it.